File: Modules/HumanResource/EmployeeProfile/MasterAllowance/MasterAllowanceForm.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace HRMSoftware.EmployeeProfile.Forms;

[FormScript("EmployeeProfile.MasterAllowance")]
[BasedOnRow(typeof(MasterAllowanceRow), CheckNames = true)]
public class MasterAllowanceForm
{
    [HalfWidth]
    public string AllowanceCode { get; set; }

    [HalfWidth]
    public double Amount { get; set; }
   
    [TextAreaEditor(Rows = 3)]
    public string Description { get; set; }




    [Category("Allowance Frequency")]
    [QuarterWidth]
    public int Recurring { get; set; }
    [QuarterWidth]
    public int OneTime { get; set; }

    [Category("Allowance Conditions")]
    public int AllowanceSubjections { get; set; }
    [QuarterWidth, HideOnInsert, HideOnUpdate]
    public int FullAttendance { get; set; }
    [QuarterWidth, HideOnInsert, HideOnUpdate]
    public int NoLate { get; set; }
    [QuarterWidth, HideOnInsert, HideOnUpdate]
    public int NoAbsence { get; set; }
    [QuarterWidth, HideOnInsert, HideOnUpdate]
    public int NoEarlyLeaving { get; set; }

    [Category("Exempted Leaves")]
    [QuarterWidth]
    public int ExemptUnpaidLeave { get; set; }
    [QuarterWidth]
    public int ExemptHospitalisationLeave { get; set; }
    [QuarterWidth]
    public int ExemptSickLeave { get; set; }
    [QuarterWidth]
    public int ExemptAnnualLeave { get; set; }
    [QuarterWidth]
    public int ExemptMaternityLeave { get; set; }
    [QuarterWidth]
    public int ExemptPaternityLeave { get; set; }
    [QuarterWidth]
    public int ExemptMarriageLeave { get; set; }
    [QuarterWidth]
    public int ExemptCompassionateLeave { get; set; }
    [QuarterWidth]
    public int ExemptEmergencyLeave { get; set; }
    [QuarterWidth]
    public int ExemptGatepassLeave { get; set; }




    [Category("Allowance Subjections")]

    public int SubjectionEis { get; set; }
    public int SubjectionEpf { get; set; }
    public int SubjectionSocso { get; set; }
    public int SubjectionPcb { get; set; }
    public int SubjectionHrdf { get; set; }
    public int SubjectionOt { get; set; }
}

File: Modules/ServerTypes/EmployeeProfile/MasterAllowanceForm.ts
import { StringEditor, DecimalEditor, TextAreaEditor, BooleanEditor, PrefixedContext } from "@serenity-is/corelib";
import { initFormType } from "@serenity-is/corelib/q";

export interface MasterAllowanceForm {
    AllowanceCode: StringEditor;
    Amount: DecimalEditor;
    Description: TextAreaEditor;
    Recurring: BooleanEditor;
    OneTime: BooleanEditor;
    AllowanceSubjections: BooleanEditor;
    FullAttendance: BooleanEditor;
    NoLate: BooleanEditor;
    NoAbsence: BooleanEditor;
    NoEarlyLeaving: BooleanEditor;
    ExemptUnpaidLeave: BooleanEditor;
    ExemptHospitalisationLeave: BooleanEditor;
    ExemptSickLeave: BooleanEditor;
    ExemptAnnualLeave: BooleanEditor;
    ExemptMaternityLeave: BooleanEditor;
    ExemptPaternityLeave: BooleanEditor;
    ExemptMarriageLeave: BooleanEditor;
    ExemptCompassionateLeave: BooleanEditor;
    ExemptEmergencyLeave: BooleanEditor;
    ExemptGatepassLeave: BooleanEditor;
    SubjectionEis: BooleanEditor;
    SubjectionEpf: BooleanEditor;
    SubjectionSocso: BooleanEditor;
    SubjectionPcb: BooleanEditor;
    SubjectionHrdf: BooleanEditor;
    SubjectionOt: BooleanEditor;
}

export class MasterAllowanceForm extends PrefixedContext {
    static formKey = 'EmployeeProfile.MasterAllowance';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!MasterAllowanceForm.init)  {
            MasterAllowanceForm.init = true;

            var w0 = StringEditor;
            var w1 = DecimalEditor;
            var w2 = TextAreaEditor;
            var w3 = BooleanEditor;

            initFormType(MasterAllowanceForm, [
                'AllowanceCode', w0,
                'Amount', w1,
                'Description', w2,
                'Recurring', w3,
                'OneTime', w3,
                'AllowanceSubjections', w3,
                'FullAttendance', w3,
                'NoLate', w3,
                'NoAbsence', w3,
                'NoEarlyLeaving', w3,
                'ExemptUnpaidLeave', w3,
                'ExemptHospitalisationLeave', w3,
                'ExemptSickLeave', w3,
                'ExemptAnnualLeave', w3,
                'ExemptMaternityLeave', w3,
                'ExemptPaternityLeave', w3,
                'ExemptMarriageLeave', w3,
                'ExemptCompassionateLeave', w3,
                'ExemptEmergencyLeave', w3,
                'ExemptGatepassLeave', w3,
                'SubjectionEis', w3,
                'SubjectionEpf', w3,
                'SubjectionSocso', w3,
                'SubjectionPcb', w3,
                'SubjectionHrdf', w3,
                'SubjectionOt', w3
            ]);
        }
    }
}